src/scoped.component.js

```diff
@@ -68,111 +66,85 @@ export class Scoped extends React.Component {
+    const namespace = isPostCss ? css.namespace : (props.namespace || Scoped.defaultNamespace)
+    const rawCss = isPostCss ? css.styles : css
 
-    if (props.css.indexOf("&") < 0 && props.css.trim().length > 0) {
-      const firstRule = props.css.substring(0, props.css.indexOf("{")).trim();
-      console.warn(
-        `Kremling's <Scoped css="..."> css prop should have the '&' character in it to scope the css classes: ${firstRule}`
-      );
+    let styleRef, kremlingAttr, kremlingAttrValue
+
+    if (!isPostCss) {
+      if (typeof css !== 'string') {
+        return
+      }
+
+      if (css.indexOf("&") < 0 && css.trim().length > 0) {
+        const firstRule = css.substring(0, props.css.indexOf("{")).trim();
+        console.warn(
+          `Kremling's <Scoped css="..."> css prop should have the '&' character in it to scope the css classes: ${firstRule}`
+        );
+      }
     }
 
-    let styleRef, kremlingAttrName, kremlingAttrValue;
-    const existingDomEl = styleTags[props.css];
+    const existingDomEl = styleTags[rawCss];
 
     if (existingDomEl) {
       styleRef = existingDomEl;
       existingDomEl.kremlings++;
-      kremlingAttrName = styleRef.kremlingAttr;
+      kremlingAttr = styleRef.kremlingAttr;
       kremlingAttrValue = styleRef.kremlingValue;
     } else {
       // The attribute for namespacing the css
-      kremlingAttrName = `data-${props.namespace || Scoped.defaultNamespace}`;
+      kremlingAttr = namespace;
       kremlingAttrValue = incrementCounter();
 
       // The css to append to the dom
-      const kremlingSelector = `[${kremlingAttrName}="${kremlingAttrValue}"]`;
-      const transformedCSS = transformCss(props.css, kremlingSelector)
+      const kremlingSelector = `[${kremlingAttr}="${kremlingAttrValue}"]`;
+      const transformedCSS = isPostCss ? rawCss : transformCss(rawCss, kremlingSelector)
 
       // The dom element
       const el = document.createElement('style');
       el.setAttribute('type', 'text/css');
       el.textContent = transformedCSS;
       el.kremlings = 1;
-      el.kremlingAttr = kremlingAttrName;
+      el.kremlingAttr = kremlingAttr;
       el.kremlingValue = kremlingAttrValue;
       document.head.appendChild(el);
-      styleTags[props.css] = el;
+      styleTags[rawCss] = el;
       styleRef = el;
     }
 
     return {
-      css: props.css,
+      isPostCss,
+      rawCss,
       styleRef,
-      kremlingAttrName,
+      kremlingAttr,
       kremlingAttrValue,
     };
   }
-
-  doneWithCss = () => {
-    if (this.state.styleRef && --this.state.styleRef.kremlings === 0) {
-      this.state.styleRef.parentNode.removeChild(this.state.styleRef);
-      delete styleTags[this.state.css];
-    }
-  }
-
-  doneWithPostcss = () => {
-    this.state.styleRef.counter -= 1;
-    if (this.state.styleRef.counter === 0) {
-      this.state.styleRef.parentNode.removeChild(this.state.styleRef);
-    }
-  }
-
-  newPostcssState = (props) => {
```

Comment:
this had a variety of bugs and it treated style elements differently than css prop or useCss does. Now that there is only one code path, it is treated in a consistent way.